main.tf

```diff
@@ -1106,6 +1333,131 @@ resource "aws_cloudwatch_event_target" "macie_alert" {
 
 
 
+resource "aws_iam_role" "security_hub_workflow" {
+  count = (var.security_hub_finding && var.enabled) ? 1 : 0
+
+  name_prefix = "marbot"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+
+  tags = var.tags
+}
+
+resource "aws_iam_role_policy" "security_hub_workflow" {
+  count = (var.security_hub_finding && var.enabled) ? 1 : 0
+
+  name_prefix = "marbot"
+  role        = join("", aws_iam_role.security_hub_workflow.*.id)
+
+  policy = <<-EOF
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": "securityhub:BatchUpdateFindings",
+        "Effect": "Allow",
+        "Resource": "*"
+      },
+      {
+        "Action": [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        "Effect": "Allow",
+        "Resource": "${join("", aws_cloudwatch_log_group.security_hub_workflow.*.arn)}"
+      }
+    ]
+  }
+  EOF
+}
+
+resource "aws_lambda_function" "security_hub_workflow" {
+  count = (var.security_hub_finding && var.enabled) ? 1 : 0
+
+  filename         = data.archive_file.lambda.output_path
+  source_code_hash = data.archive_file.lambda.output_base64sha256
+  function_name    = "marbot-basic-security-hub-finding-workflow-${random_id.id8.hex}"
```

Comment:
Does `random_id` create a random ID only once? Or does this re-create the resource on every `terraform apply`? Could not find any hints in the documentation. Did you test that?

Reply:
the random id stays stable once generated.